java/ql/src/Security/CWE/CWE-036/OpenStream.ql

```diff
@@ -0,0 +1,50 @@
+import java
+import semmle.code.java.dataflow.TaintTracking
+import semmle.code.java.dataflow.FlowSources
+import DataFlow::PathGraph
+
+class URLConstructor extends ClassInstanceExpr {
+  URLConstructor() { this.getConstructor().getDeclaringType().getQualifiedName() = "java.net.URL" }
```

Comment:
I think there's already a declared type somewhere for this type?

If not, you may want to consider declaring one in this file:
https://github.com/Semmle/ql/blob/1d4f894583621ed16bf293086527320f80a957ae/java/ql/src/semmle/code/java/frameworks/Networking.qll

Reply:
Yeah, looks like a good idea to include a java.net.URL Type in that file.